src/MergeContext.js

```diff
@@ -449,6 +449,22 @@ class BranchPosition
 
     async compute() {
         this._status = await GH.compareCommits(this._baseRef, this._featureRef);
+        return this._status;
+    }
+
+    async computeUntilAhead() {
+        const desiredStatus = "ahead";
+        // TODO: Stop (poorly) duplicating these Util.sleep() loops.
+        const longestSleep = 16 * 1000; // ms; ~30 seconds overall
+        let nextSleep = 0;
+        while (await this.compute() !== desiredStatus) {
+            if (nextSleep >= longestSleep)
+                throw new Error(`failed to reach the desired branch state; wanted ${desiredStatus} but got ${this._status} despite waiting for ${nextSleep/1000} seconds`);
```

Comment:
Probably the overall spent time would be more informative - not just the last step duration. Basically, it should be lastStep*2-1. Do you think we should adjust it?

Reply:
> Probably the overall spent time would be more informative - not just the last step duration. 

I agree.

> Basically, it should be lastStep*2-1. 

Hm.. Is not it `(2^n-1)*a`, where n is the number of sleep calls, and a is the first sleep duration? For example, if we slept three times (for a, 2a, and 4a seconds), then the total for this n=3 case is 7a rather than `8a-1`.

> Do you think we should adjust it?

Yes, please, provided there is a simple way to report the actual total delay rather than the sum of Sleep() parameter values. AFAICT, that sum may not, in some cases, represent the actual time we delayed progress, even approximately, because our sleep might be interrupted and/or because compute() itself may take time.

Reply:
Done at abfd49d.

Reply:
LGTM. Please feel free to merge and deploy.